src/rating-indicator.scss

```diff
@@ -0,0 +1,220 @@
+@import "./new-settings";
+@import "./mixins";
+@import "./icon";
+
+$block: #{$fd-namespace}-rating-indicator;
+
+.#{$block} {
+  $fd-rating-indicator-sizes: (
+    "cozy": ("fontSize": 1.5rem, "spacingBetweenIcons": 0.1875rem, "controlMargin": 0.8125rem 0),
+    "compact": ("fontSize": 1rem, "spacingBetweenIcons": 0.125rem, "controlMargin": 0.5rem 0),
+    "condensed": ("fontSize": 1rem, "spacingBetweenIcons": 0.125rem),
+    "xs": ("fontSize": 0.75rem, "spacingBetweenIcons": 1px),
+    "s": ("fontSize": 1rem, "spacingBetweenIcons": 0.125rem),
+    "m": ("fontSize": 1.375rem, "spacingBetweenIcons": 0.1875rem),
+    "l": ("fontSize": 2rem, "spacingBetweenIcons": 0.25rem)
+  );
+  $fd-rating-indicator-default-size: map-get($fd-rating-indicator-sizes, 'm');
+  $fd-rating-indicator-default-font-size: map_get($fd-rating-indicator-default-size, "fontSize");
+  $fd-rating-indicator-default-spacing-between-icons: map_get($fd-rating-indicator-default-size, "spacingBetweenIcons");
+
+  @mixin fd-rating-indicator-with-half-star($iconFontSize, $spacingBetweenIcons) {
+    .#{$block}__input {
+      width: $iconFontSize / 2;
+      height: $iconFontSize;
+    }
+
+    .#{$block}__label {
+      margin-right: 0;
+
+      &:nth-child(4n+4) {
+        margin-right: $spacingBetweenIcons;
+        justify-content: flex-end;
+      }
+
+      overflow: hidden;
+      width: $iconFontSize / 2;
+    }
+  }
+
+  @include fd-reset();
+
+  position: relative;
+
+  &__input {
+    @include fd-reset();
+
+    position: absolute;
+    opacity: 0.0001;
+    z-index: -1;
+    top: 0;
+    width: $fd-rating-indicator-default-font-size;
+    height: $fd-rating-indicator-default-font-size;
+
+    &:checked {
+      @include fd-fiori-focus(0) {
+        + .#{$block}__label {
+          outline-offset: 0;
+          outline-width: var(--sapContent_FocusWidth);
+          outline-color: var(--sapContent_FocusColor);
+          outline-style: var(--sapContent_FocusStyle);
+        }
+      };
+
+      ~ input + .#{$block}__label {
+        @include fd-icon-glyph('unfavorite') {
+          color: var(--sapRating_Indicator_Unselected_Icon_Color);
+        }
+
+        &:hover::before {
+          opacity: 1;
+        }
+      }
+    }
+  }
+
+  &__label {
+    @include fd-reset();
+    @include action-cursor();
+
+    @include fd-icon-glyph('favorite') {
+      font-family: "SAP-icons";
+      color: var(--sapRating_Indicator_Selected_Icon_Color);
+      font-size: $fd-rating-indicator-default-font-size;
+    }
+
+    display: flex;
+    align-items: center;
+    float: left;
```

Comment:
What about adding `display: flex` to `fd-rating-indicator__container`, to get rid of floats?

Reply:
Impossible due to accessibiliyt stuff